pkg/filebacked/file.go

```diff
@@ -72,23 +83,43 @@ func (w *Writer) Reader() (reader *Reader) {
 	err := w.file.Sync()
 	if err == nil {
 		err = os.Link(w.path, path)
+		if err != nil {
+			log.Error(err, "link failed.")
+		}
 	}
 	reader = &Reader{
 		error: liberr.Wrap(err),
 		path:  path,
 	}
+	runtime.SetFinalizer(
+		reader,
+		func(r *Reader) {
+			r.Close()
+		})
+	log.V(1).Info(
+		"writer: reader created.",
+		"path",
+		w.path,
+		"link",
+		reader.path)
 
 	return
 }
 
 //
 // Close the writer.
 func (w *Writer) Close() {
-	if w.file != nil {
-		_ = w.file.Close()
+	defer func() {
 		_ = os.Remove(w.path)
```

Comment:
Make sure the hard-link is deleted even if the reader (iterator) is never read.  Eg: Next() is never called.